Core/Src/DataLoggingTask.c

```diff
@@ -1,17 +1,33 @@
+/**
+ * @file DataLoggingTask.c
+ * 
+ * Task in charge of logging data to the SD card as an offline backup 
+ * to the data sent over ethernet to Data Acquisition.
+ * Data is added to the logging queue by the Data Reading Task.       
+ * 
+ * @copyright Copyright (c) 2022 UT Longhorn Racing Solar
+ * 
+ * TODO: error check
+ */
+
 #include "Tasks.h"
 #include "SDCard.h"
+#include "config.h"
 
 #define MOUNTCYCLES 500
 
 void DataLoggingTask(void* argument){
     int cntr = 0;
+    if (SDCard_Init() != FR_OK);
+    xSemaphoreTake(InitSem, 0);
+    while(uxSemaphoreGetCount(InitSem) != 0);
 
     while (1){
         if (SDCard_Sort_Write_Data() == FR_OK) cntr++; //increment counter if data was written
         if (cntr > MOUNTCYCLES) {
-            if (SDCard_CloseFileSystem() != FR_OK); //TODO: ERROR CHECKING HERE
+            if (SDCard_CloseFileSystem() != FR_OK);
```

Comment:
I would get rid of the if statement and add a comment saying if the error occurs, there isn't much we can do about it.

Reply:
I could add a function that spins somewhere, like the BPS fault state. Maybe I could make use of the LEDs on the board?

Reply:
That would be nice. We can save it for another PR.